core/src/test/java/org/elasticsearch/cluster/routing/allocation/MaxRetryAllocationDeciderTests.java

```diff
@@ -208,4 +213,45 @@ public void testFailedAllocation() {
         assertEquals(routingTable.index("idx").shard(0).shards().get(0).state(), INITIALIZING);
         assertEquals(routingTable.index("idx").shard(0).shards().get(0).unassignedInfo().getMessage(), "ZOOOMG");
     }
+
+    /**
+     * Tests that {@link MaxRetryAllocationDecider#canForceAllocatePrimary(ShardRouting, RoutingNode, RoutingAllocation)} adheres
+     * to the maximum retry count and returns a NO decision if the maximum number of retries is exceeded.
+     */
+    public void testCanForceAllocatePrimary() {
```

Comment:
If you just want to check that `canForceAllocatePrimary` adheres to the maximum retry count, isn't it simpler to extend the existing test `testFailedAllocation` and 
do a simple check there? Such a check can be made by selecting the unassigned shard in the routing table and doing 
```
assertEquals(Decision.NO, new MaxRetryAllocationDecider(Settings.EMPTY).canForceAllocatePrimary(shardRouting, null, new RoutingAllocation(null, null, clusterState, null, 0, false)))
```